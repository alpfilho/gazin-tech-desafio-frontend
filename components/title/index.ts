export { Title } from './title.styles';
