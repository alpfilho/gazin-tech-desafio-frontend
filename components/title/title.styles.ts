import styled from 'styled-components';

export const Title = styled.h1`
	padding: 0;
	margin: 0;

	font-family: 'Poppins', sans-serif;
	font-weight: 700;

	user-select: none;

	font-size: 2.5em;
	padding-bottom: 0.25em;

	color: ${({ theme }) => theme.text};

	@media screen and (max-width: 768px) {
		font-size: 2em;
	}
`;
