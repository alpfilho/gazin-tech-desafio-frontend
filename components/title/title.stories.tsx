import { ComponentMeta, ComponentStory } from '@storybook/react';

import { Title } from './title.styles';

export default {
	title: 'Components/Title',
	component: Title
} as ComponentMeta<typeof Title>;

export const Default: ComponentStory<typeof Title> = () => <Title>Example Title</Title>;
