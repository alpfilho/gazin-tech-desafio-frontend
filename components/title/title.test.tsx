import { render, screen } from '@testing-library/react';
import { composeStories } from '@storybook/testing-react';

import * as stories from './title.stories';

const { Default } = composeStories(stories);

describe('Title', () => {
	it('render a title', () => {
		render(<Default />);

		const exampleContent = screen.getByText('Example Title');
		expect(exampleContent).toBeInTheDocument();
	});

	it('title render unchanged', () => {
		const { container } = render(<Default />);

		expect(container).toMatchSnapshot();
	});
});
