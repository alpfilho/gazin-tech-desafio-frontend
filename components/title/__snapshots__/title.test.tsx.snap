// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title title render unchanged 1`] = `
<div>
  <h1
    class="Title-sc-12jcw8v-0 bJXsoN"
  >
    Example Title
  </h1>
</div>
`;
